src/CPUActivityAgent.cpp

```diff
@@ -324,19 +324,14 @@ namespace geopm
             /////////////////////////////////////////////
             // L3 Total External Bandwidth Measurement //
             /////////////////////////////////////////////
-            auto qm_max_itr = m_qm_max_rate.lower_bound(uncore_freq);
-            // lower_bound returns the first element that does not compare
-            // less than the uncore_freq, which may cause issues when
-            // the uncore_freq signal does not exactly match the keys in the
-            // map.
-            //
-            // To handle this case we grab an entry one less than the
-            // lower bound if and only if the qm_max_itr isn't the first key
-            // and the uncore_freq is not an exact match.
-            if(qm_max_itr != m_qm_max_rate.begin() &&
-               m_qm_max_rate.find(uncore_freq) == m_qm_max_rate.end()) {
+            // Get max mem. bandwidth for uncore_freq. There may be uncore
+            // frequencies for which an exact match doesn't exist. To handle
+            // this case, we grab the entry prior to upper_bound() (as long as
+            // it's not the first entry), in other words, the last entry that
+            // is <= uncore_freq.
+            auto qm_max_itr = m_qm_max_rate.upper_bound(uncore_freq);
+            if (qm_max_itr != m_qm_max_rate.begin())
                 qm_max_itr--;
```

Comment:
```suggestion
            if (qm_max_itr != m_qm_max_rate.begin()) {
                --qm_max_itr;
            }
```

or on one line: `if (qm_max_itr != m_qm_max_rate.begin()) --qm_max_itr;`

Not a required change, since this feels like a style preference issue and I don't know if the team has any rules about this. My personal motivation is that I've seen bugs appear on projects from moving code into or out the body of this style of if-statement during later changes.

Reply:
Yeah, I don't actually need post-increment semantics here (was mainly following common style). In the rest of the codebase, for similar operations, we have a bit of a mix, but in the vast majority of cases we use pre-increment. I'll update this accordingly.